File: C#/myLeetCode/String/Offer 58 - II. 左旋转字符串.cs
// 剑指 Offer 58 - II. 左旋转字符串
// https://leetcode.cn/problems/zuo-xuan-zhuan-zi-fu-chuan-lcof/

public class Solution {
    public string ReverseLeftWords(string s, int n) {
        return s.Substring(n) + s.Substring(0, n);
    }
}
public class Program {
    public static void Main() {
        Solution solution = new Solution();

        string input = "abcdefg";
        int n = 2;
        string result = solution.ReverseLeftWords(input, n);

        Console.WriteLine($"Input: '{input}' with n={n}");
        Console.WriteLine($"Output: '{result}'");
    }
}

File: C++/代码随想录/String/Offer 58 - II. 左旋转字符串.cpp
// 剑指 Offer 58 - II. 左旋转字符串
// https://leetcode.cn/problems/zuo-xuan-zhuan-zi-fu-chuan-lcof/

#include <iostream>
#include <algorithm>

class Solution {
public:
    std::string reverseLeftWords(std::string str, int n) {
        std::rotate(str.begin(), str.begin() + n, str.end());
        return str;
    }
};

int main() {
    Solution sol;
    std::string str = "abcdefg";
    int n = 2;
    std::cout << sol.reverseLeftWords(str, n) << std::endl;
    return 0;
}
